xdsl/transforms/loop_invariant_code_motion.py

```diff
@@ -32,9 +41,82 @@ def can_be_hoisted(op: Operation, target_region: Region) -> bool | None:
     for child in op.walk():
         for operand in child.operands:
             operand_owner = operand.owner
-            assert operand_owner is not None
             if op.is_ancestor(operand_owner):
                 continue
             if target_region.is_ancestor(operand_owner):
                 return False
     return True
+
+
+def _move_loop_invariant_code(
+    regions: Sequence[Region],
```

Comment:
Is there any case where we cannot handle this region by region? I would just take a single region as argument here no?

Reply:
I think this is there for loop ops with multiple regions, maybe like a while loop?

Reply:
What I mean is that in that case, we should probably write `for region in op.regions: _move_loop_invariant_code(region)`, which kind of makes the code simpler here I think

Reply:
Done